### ISSUE TITLE
Figures in block2 environment for PDF output result in an error

### ISSUE THREAD

**Issue by florisvdh** [rstudio/bookdown#248]
I get the following error from LaTeX:

`LaTeX Error: Not in outer par mode.`

It results from inserting a figure _inside_ a block2 chunk, like this: 

`![(\#fig:tux)*This is Tux, the Linux mascot.*](Tux.png)`

However in HTML output this poses no problem. I attached an [RStudio project](https://github.com/rstudio/bookdown/files/605935/bookdowntest.zip) to reproduce this, both for html and pdf.

**Comment by florisvdh** [comment]
update: I made a correction to the Rstudio project link [above](https://github.com/rstudio/bookdown/issues/248#issue-190946449) - a wrong "header.tex" file was included. (This is no remedy to the issue of course.)

**Comment by yihui** [comment]
This is an issue with the `mdframed` environment. If I replace `type='block'` with `type='small'`, the issue is gone. You may reduce this example to a minimal LaTeX example, and ask on StackExchange.

**Comment by florisvdh** [comment]
Thank you! I will look further into these things later on.

**Comment by florisvdh** [comment]
`type='small'` does not result in a box, as this environment is not defined. I think that's the reason why the issue was gone. As this approach returns no box anymore, it does not appear to be the solution.

I've also compared with `minipage` environments instead of `mdframed` to produce coloured boxes ([link on StackExchange](https://tex.stackexchange.com/questions/29769/colored-box-in-new-environment#29772)). This worked to produce boxes, but it did not solve the problem of inserting figures.

So I sticked with the `mdframed` environment for now, and I found [**this solution**](https://tex.stackexchange.com/questions/124346/latex-error-not-in-outer-par-mode#130770) on StackExchange. It comes down to using a `minipage` environment and directly use `\includegraphics`. 

Unzipping [this tex file](https://github.com/rstudio/bookdown/files/754370/_main.tex.zip) into the Rstudio project provided [above](https://github.com/rstudio/bookdown/issues/248#issue-190946449) should give the desired output. See lines 27 and 143-147 to see the specific things that have been altered, compared to figures outside of a block.

Perhaps invoking the `minipage` environment in this way, instead of using `\begin{figure}` and `\end{figure}`, may be achieved by the bookdown package? This could solve the problem of showing figures inside blocks in PDF output. I expect that it would work inside and outside of block environments.
